File: 268.missing-number.py
# @leet start
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums) 
        map = [False for _ in range(n+1)]
        
        for num in nums:
            map[num] = True
        
        for i in range(n+1):
            if not map[i]:
                return i
        return 0
# @leet end

File: 268.missing-number.cpp
// @leet start
#include <algorithm>
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> d_nums;

        for (int i : nums) {
            d_nums.insert(i);
        }

        for (int i = 0; i <= n; i++) {
            if (d_nums.find(i) == d_nums.end()) {
                return i;
            }
        }

        return -1;
    }
};
// @leet end
